src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,450 @@
+    continuous_linear_map.coe_mul, function.comp_app, continuous_linear_map.map_smul]})
+  (λ k a b, by {ext; simp only [mul_fst, smul_fst, mul_snd, smul_snd, algebra.mul_smul_comm,
+    coe_smul', continuous_linear_map.coe_mul, pi.smul_apply, function.comp_app]})
+
+/-!
+### Star structure
+-/
+
+section star
+
+variables [star_ring 𝕜] [star_ring A] [star_module 𝕜 A] [normed_star_group A]
+
+/-- The star operation on `a : 𝓜(𝕜, A)` is given by
+`(star a).to_prod = (star ∘ a.snd ∘ star, star ∘ a.fst ∘ star)`. -/
+instance : has_star 𝓜(𝕜, A) :=
+{ star := λ a,
+  { fst := (((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A).comp a.snd).comp
+      ((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A),
+    snd := (((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A).comp a.fst).comp
+      ((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A),
+    central := λ x y, by simpa only [star_mul, star_star]
+      using (congr_arg star (a.central (star y) (star x))).symm } }
+
+@[simp] lemma star_fst (a : 𝓜(𝕜, A)) (b : A) : (star a).fst b = star (a.snd (star b)) := rfl
+@[simp] lemma star_snd (a : 𝓜(𝕜, A)) (b : A) : (star a).snd b = star (a.fst (star b)) := rfl
+
+instance : star_add_monoid 𝓜(𝕜, A) :=
+{ star_involutive := λ x, by {ext; simp only [star_fst, star_snd, star_star]},
+  star_add := λ x y, by {ext; simp only [star_fst, star_snd, add_fst, add_snd,
+    continuous_linear_map.add_apply, star_add]},
+  .. double_centralizer.has_star }
+
+instance : star_ring 𝓜(𝕜, A) :=
+{ star_mul := λ a b, by {ext; simp only [star_fst, star_snd, mul_fst, mul_snd, star_star,
+    continuous_linear_map.coe_mul, function.comp_app]},
+  .. double_centralizer.star_add_monoid }
+
+instance : star_module 𝕜 𝓜(𝕜, A) :=
+{ star_smul := λ k a, by {ext; exact star_smul _ _},
+  .. double_centralizer.star_add_monoid }
+
+end star
+
+/-!
+### Coercion from an algebra into its multiplier algebra
+-/
+
+section
+variables [star_ring 𝕜] [star_ring A] [star_module 𝕜 A] [normed_star_group A]
+
+/-- The coercion of an algebra into its multiplier algebra as a non-unital star algebra
+homomorphism. -/
+def coe_hom : A →⋆ₙₐ[𝕜] 𝓜(𝕜, A) :=
+{ to_fun := λ a,
+  { fst := continuous_linear_map.mul 𝕜 A a,
+    snd := (continuous_linear_map.mul 𝕜 A).flip a,
+    central := λ x y, mul_assoc _ _ _ },
+  map_smul' := λ k a, by {ext; simp only [continuous_linear_map.map_smul, smul_fst, smul_snd]},
+  map_zero' := by {ext; simp only [map_zero, zero_fst, zero_snd]},
+  map_add' := λ a b, by {ext; simp only [map_add, add_fst, add_snd]},
+  map_mul' := λ a b, by {ext; simp only [mul_apply', flip_apply, mul_fst, mul_snd,
+    continuous_linear_map.coe_mul, function.comp_app, mul_assoc]},
+  map_star' := λ a, by {ext; simp only [mul_apply', star_fst, star_snd, flip_apply,
+    star_mul, star_star]} }
+end
+
+noncomputable instance [star_ring 𝕜] [star_ring A] [star_module 𝕜 A] [normed_star_group A] :
+  has_coe_t A 𝓜(𝕜, A) :=
+{ coe := (double_centralizer.coe_hom : A → 𝓜(𝕜, A)) }
```

Comment:
If you swap the order then you can weaken the typeclass assumptions:
```suggestion
-- TODO: move this further up the file since it doesn't need these typeclass assumptions
/-- The coercion of an algebra into its multiplier algebra by setting `L` and `R` as left- and right- multiplication. -/
noncomputable instance [star_ring 𝕜] [star_ring A] [star_module 𝕜 A] [normed_star_group A] :
  has_coe_t A 𝓜(𝕜, A) :=
{ coe := λ a,
  { fst := continuous_linear_map.mul 𝕜 A a,
    snd := (continuous_linear_map.mul 𝕜 A).flip a,
    central := λ x y, mul_assoc _ _ _ } }

/-- `coe` as a non-unital star algebra homomorphism. -/
@[simps]
def coe_hom : A →⋆ₙₐ[𝕜] 𝓜(𝕜, A) :=
{ to_fun := coe,
  map_smul' := λ k a, by {ext; simp only [continuous_linear_map.map_smul, smul_fst, smul_snd]},
  map_zero' := by {ext; simp only [map_zero, zero_fst, zero_snd]},
  map_add' := λ a b, by {ext; simp only [map_add, add_fst, add_snd]},
  map_mul' := λ a b, by {ext; simp only [mul_apply', flip_apply, mul_fst, mul_snd,
    continuous_linear_map.coe_mul, function.comp_app, mul_assoc]},
  map_star' := λ a, by {ext; simp only [mul_apply', star_fst, star_snd, flip_apply,
    star_mul, star_star]} }
end
```

Reply:
That's actually what I had originally, but then I switched it because I thought Lean 4 would be able to take advantage of it this way since it unfolds coercions. I'll change it back for now and we can address that in Lean 4 if we want.